types/apostrophe/index.d.ts

```diff
@@ -0,0 +1,302 @@
+            relationshipsField: any,
+            objectField: any,
+            options: any,
+            callback: () => any
+        ): void;
+        join(
+            req: any,
+            schema: Schema,
+            objectOrArray: any,
+            withJoins: any,
+            callback: () => any
+        ): void;
+        addFieldType(type: AposType): void;
+        getFieldType(typeName: string): void;
+        addFilters(schema: Schema, options: any, cursor: any): void;
+        joinFilterChoices(field: any, cursor: any, valueField: any): void;
+        addJoinSlugFilter(field: any, cursor: any, suffix: any): void;
+        pageServe(req: any): void;
+        sortedDistinct(property: any, cursor: any, callback: () => any): void;
+        cursorFilterInterested(cursor: any, name: string): void;
+        afterInit(): void;
+        validate(schema: Schema, options: any): void;
+    }
+
+    interface AposModule {
+        emit(name: string): void;
+        on(name: string, methodName: any, fn: () => any): void;
+    }
+
+    type AposCoreModules =
+        | "apostrophe-admin-bar"
+        | "apostrophe-any-page-manager"
+        | "apostrophe-areas"
+        | "apostrophe-assets"
+        | "apostrophe-attachments"
+        | "apostrophe-browser-utils"
+        | "apostrophe-caches"
+        | "apostrophe-custom-pages"
+        | "apostrophe-db"
+        | "apostrophe-doc-type-manager"
+        | "apostrophe-docs"
+        | "apostrophe-email"
+        | "apostrophe-express"
+        | "apostrophe-files"
+        | "apostrophe-files-widgets"
+        | "apostrophe-global"
+        | "apostrophe-groups"
+        | "apostrophe-html-widgets"
+        | "apostrophe-i18n"
+        | "apostrophe-images"
+        | "apostrophe-images-widgets"
+        | "apostrophe-jobs"
+        | "apostrophe-launder"
+        | "apostrophe-locks"
+        | "apostrophe-login"
+        | "apostrophe-migrations"
+        | "apostrophe-modal"
+        | "apostrophe-module"
+        | "apostrophe-notifications"
+        | "apostrophe-oembed"
+        | "apostrophe-pager"
+        | "apostrophe-pages"
+        | "apostrophe-permissions"
+        | "apostrophe-pieces"
+        | "apostrophe-pieces-pages"
+        | "apostrophe-pieces-widgets"
+        | "apostrophe-polymorphic-manager"
+        | "apostrophe-push"
+        | "apostrophe-rich-text-widgets"
+        | "apostrophe-schemas"
+        | "apostrophe-search"
+        | "apostrophe-service-bridge"
+        | "apostrophe-soft-redirects"
+        | "apostrophe-tags"
+        | "apostrophe-tasks"
+        | "apostrophe-templates"
+        | "apostrophe-ui"
+        | "apostrophe-urls"
+        | "apostrophe-users"
+        | "apostrophe-utils"
+        | "apostrophe-versions"
+        | "apostrophe-video-fields"
+        | "apostrophe-video-widgets"
+        | "apostrophe-widgets";
+
+    // Pass in custom modules to AposModuleOptions to allow them in extend
+    interface AposModuleOptions<C> {
```

Comment:
Consider making this type parameter optional (i.e. `C = {}`).

Reply:
Ahh thank you! I could not find the syntax for making that parameter optional!